tests/apps/query_performing_app/apps.py

```diff
@@ -0,0 +1,56 @@
+from django.apps import AppConfig
+from django.db import connections
+
+
+class BaseAppConfig(AppConfig):
+    name = "apps.query_performing_app"
+    database = "default"
+    use_model = True
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.query_results = []
+
+    def ready(self):
+        self.query_results = []
+
+        if self.use_model:
+            self._perform_query_using_model()
+        else:
+            self._perform_query_using_cursor()
+
+    def _perform_query_using_model(self):
+        from ..models import TotallyNormal
+
+        queryset = TotallyNormal.objects.using(self.database)
+        queryset.create(name=self.label)
+        self.query_results = list(queryset.filter(name=self.label))
+
+    def _perform_query_using_cursor(self):
+        from ..models import TotallyNormal
+
+        with connections[self.database].cursor() as cursor:
+            db_table = TotallyNormal._meta.db_table
+            cursor.execute(f"INSERT INTO {db_table}(name) VALUES (%s)", [self.label])
+            cursor.execute(f"SELECT name FROM {db_table} WHERE name = %s", [self.label])
+            self.query_results = cursor.fetchall()
```

Comment:
I'd refactored out two subclasses instead of using flags, e.g.
```python
from django.apps import AppConfig
from django.db import connections


class BaseAppConfig(AppConfig):
    name = "apps.query_performing_app"
    database = "default"

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.query_results = []

    def ready(self):
        self.query_results = []
        self.perform_query()
            
            
class CursorQueryAppConfig(BaseAppConfig):
    def perform_query(self):
        from ..models import TotallyNormal
    
        with connections[self.database].cursor() as cursor:
            db_table = TotallyNormal._meta.db_table
            cursor.execute(f"INSERT INTO {db_table}(name) VALUES (%s)", [self.label])
            cursor.execute(f"SELECT name FROM {db_table} WHERE name = %s", [self.label])
            self.query_results = cursor.fetchall()
            
            
class ModelQueryAppConfig(BaseAppConfig):
     def perform_query(self):
        from ..models import TotallyNormal
     
        queryset = TotallyNormal.objects.using(self.database)
        queryset.create(name=self.label)
        self.query_results = list(queryset.filter(name=self.label))
        
            
class QueryDefaultDatabaseUsingModelAppConfig(ModelQueryAppConfig):
    database = "default"
    
    
class QueryOtherDatabaseUsingModelAppConfig(ModelQueryAppConfig):
    database = "other"
    
    
class QueryDefaultDatabaseUsingCursorAppConfig(CursorQueryAppConfig):
    database = "default"
    
    
class QueryOtherDatabaseUsingCursorAppConfig(CursorQueryAppConfig):
    database = "other"
```